docs/internal-states.md

```diff
@@ -0,0 +1,178 @@
+# Windmill Internal State 
```

Comment:
path should probably be in the how-tos folder and titled:

"How to persist state across flow step executions"